lib/MeshBVHNode.js

```diff
@@ -12,11 +12,8 @@ class MeshBVHNode {
 
 	constructor() {
 
-		this.boundingData = null;
-		this.geometry = null;
-		this.tris = null;
-		this.children = null;
-		this.splitAxis = - 1;
+		// internal nodes have boundingData, children, and splitAxis
```

Comment:
I'm wondering if there's a performance trade-off here by not defining all the fields on every node. Here's a benchmark that looks into the performance implications of using differently shaped objects:
https://github.com/davidmarkclements/v8-perf#polymorphic-vs-monomorphic-code

Reply:
Yeah, having an extra type in the inline caches must cost something, but the tree size difference when half the nodes only need half the fields and half only need the other fields is so much that I suspect it must be faster (i.e. due to more nodes fitting in a page.) I didn't carefully measure it, though, because even if it were a little slower I would still want to do it to cut the memory usage.

Reply:
I did a little bit of testing and didn't see any significant changes so this seems okay.

If you really wanted to save more memory you could store `count` and `offset` in a more optimally sized (`UInt16Array`, `UInt32Array`, etc) array based on the length of the index array.

Reply:
True. But the long-term approach for making things more efficient from here is to serialize the whole output tree into one big typed array, so this intermediate step isn't super appealing to me.